pkg/tracing/jaeger/config_yaml.go

```diff
@@ -0,0 +1,213 @@
+package jaeger
+
+import (
+	"fmt"
+	"net/url"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/opentracing/opentracing-go"
+	"github.com/pkg/errors"
+	"github.com/uber/jaeger-client-go"
+	"github.com/uber/jaeger-client-go/config"
+	"gopkg.in/yaml.v2"
+)
+
+type Config struct {
+	ServiceName            string `yaml:"service_name"`
+	Disabled               string `yaml:"disabled"`
+	RPCMetrics             string `yaml:"rpc_metrics"`
+	Tags                   string `yaml:"tags"`
+	SamplerType            string `yaml:"sampler_type"`
+	SamplerParam           string `yaml:"sampler_param"`
+	SamplerManagerHostPort string `yaml:"sampler_manager_host_port"`
+	SamplerMaxOperations   string `yaml:"sampler_max_operations"`
+	SamplerRefreshInterval string `yaml:"sampler_refresh_interval"`
+	ReporterMaxQueueSize   string `yaml:"reporter_max_queue_size"`
+	ReporterFlushInterval  string `yaml:"reporter_flush_interval"`
+	ReporterLogSpans       string `yaml:"reporter_log_spans"`
+	Endpoint               string `yaml:"endpoint"`
+	User                   string `yaml:"user"`
+	Password               string `yaml:"password"`
+	AgentHost              string `yaml:"agent_host"`
+	AgentPort              string `yaml:"agent_port"`
+}
+
+// FromYaml uses config YAML to set the tracer's Configuration.
+func FromYaml(cfg []byte) (*config.Configuration, error) {
+	conf := &Config{}
+
+	if err := yaml.Unmarshal(cfg, &conf); err != nil {
+		return nil, err
+	}
+
+	c := &config.Configuration{}
+
+	if e := conf.ServiceName; e != "" {
+		c.ServiceName = e
+	}
+
+	if e := conf.RPCMetrics; e != "" {
+		if value, err := strconv.ParseBool(e); err == nil {
```

Comment:
Couldn't we make a field bool and get rid of parsing?

Reply:
Good point, missed that - We can enforce YAML unmarshalling to parse that for us.